cmd/jaeger/config-docs/command.go

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package configdocs
+
+import (
+	"encoding/json"
+	"fmt"
+	"go/ast"
+	"go/token"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal"
+	"github.com/spf13/cobra"
+	"go.opentelemetry.io/collector/component"
+	"golang.org/x/tools/go/packages"
+)
+
+// FieldDoc represents documentation for a struct field.
+type FieldDoc struct {
+	Name         string      `json:"name"`
+	Type         string      `json:"type"`
+	Tag          string      `json:"tag,omitempty"`
+	DefaultValue interface{} `json:"default_value,omitempty"`
+	Comment      string      `json:"comment,omitempty"`
+}
+
+// StructDoc represents documentation for a struct.
+type StructDoc struct {
+	Name    string     `json:"name"`
+	Fields  []FieldDoc `json:"fields"`
+	Comment string     `json:"comment,omitempty"`
+}
+
+// Command returns the config-docs command.
+func Command() *cobra.Command {
+	return &cobra.Command{
+		Use:   "config-docs",
+		Short: "Extracts and generates configuration documentation from structs",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return GenerateDocs()
+		},
+	}
+}
+
+func GenerateDocs() error {
+	// List of target directories for AST parsing
+	targetPackages := []string{
+		"github.com/jaegertracing/jaeger/cmd/jaeger/internal/exporters/storageexporter",
+		"github.com/jaegertracing/jaeger/pkg/es/config",
+		"go.opentelemetry.io/otel",
+	}
+
+	schema := map[string]interface{}{
+		"$schema":     "https://json-schema.org/draft/2020-12/schema",
+		"type":        "object",
+		"definitions": map[string]interface{}{},
+	}
+
+	// Load packages using `go/packages`
+	cfg := &packages.Config{
+		Mode: packages.NeedSyntax | packages.NeedTypes | packages.NeedTypesInfo | packages.NeedFiles,
+	}
+	pkgs, err := packages.Load(cfg, targetPackages...)
+	if err != nil {
+		log.Fatalf("Error loading packages: %v", err)
+	}
+
+	// Extract struct definitions from AST
+	for _, pkg := range pkgs {
+		for _, syntax := range pkg.Syntax {
+			structs := parseAST(syntax)
+			for _, structDoc := range structs {
+				schema["definitions"].(map[string]interface{})[structDoc.Name] = structToSchema(structDoc)
+			}
+		}
+	}
+
+	// Extract referenced struct names from `components.go`
+	referencedStructs := extractStructsFromComponents()
```

Comment:
I recommend better organization for the code, in separate files
  * command itself
  * traversal that locates needed structs
  * output data model
  * output generator from parsed structs